manifests/service.pp

```diff
@@ -35,14 +35,38 @@
     }
   }
 
-  service { "${foreman_service}.socket":
-    ensure => $foreman_service_ensure,
-    enable => $foreman_service_enable,
-  }
+  if $foreman::deployment_mode == 'package' {
+    service { "${foreman_service}.socket":
+      ensure => $foreman_service_ensure,
+      enable => $foreman_service_enable,
+    }
 
-  service { $foreman_service:
-    ensure => $foreman_service_ensure,
-    enable => $foreman_service_enable,
-    before => Service["${foreman_service}.socket"],
+    service { $foreman_service:
+      ensure => $foreman_service_ensure,
+      enable => $foreman_service_enable,
+      before => Service["${foreman_service}.socket"],
+    }
+  } else {
+    podman::quadlet { 'foreman.container':
+      ensure          => present,
+      unit_entry      => {
+        'Description' => 'Foreman',
+      },
+      service_entry   => {
+        'TimeoutStartSec' => '900',
+      },
+      container_entry => {
+        'Image'         => 'quay.io/evgeni/foreman-rpm:latest',
+        'Volume'        => ['/etc/foreman/:/etc/foreman/'],
+        'AddCapability' => ['CAP_DAC_OVERRIDE', 'CAP_IPC_OWNER'],
```

Comment:
the `foreman` user inside the container might have a different UID than outside, fucking up file access :/

(I am not sure this is overall a good idea, but here we are for now)

Reply:
Feels like a nasty hack that we need to resolve prior to going production with this.